src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -235,6 +235,8 @@ int module_set_configuration(struct processing_module *mod,
 			     enum module_cfg_fragment_position pos, size_t data_offset_size,
 			     const uint8_t *fragment, size_t fragment_size, uint8_t *response,
 			     size_t response_size);
+int module_bind(struct processing_module *mod, void *data);
```

Comment:
Could we have some more specific type than "void *data"? Given this is already "generic.h", we should know what is passed. If there's some other variant of data passed, then that should be probably another ops.